src/Html.php

```diff
@@ -1384,21 +1549,49 @@ public static function ol($items, array $options = []): string
      * call. This method will take out these elements, if any: "prompt", "options" and "groups". See more details in
      * {@see dropDownList()} for the explanation of these elements.
      *
+     * @psalm-param iterable<array-key, string>|string|null $selection
+     * @psalm-param SelectItems $items
+     * @psalm-param SelectHtmlOptions $tagOptions
+     *
      * @throws JsonException
      *
      * @return string the generated list options
      */
-    public static function renderSelectOptions($selection, array $items, array &$tagOptions = []): string
+    public static function renderSelectOptions($selection, array $items, array $tagOptions = []): string
+    {
+        return self::renderSelectOptionTags($selection, $items, $tagOptions);
+    }
+
+    /**
+     * @param array|bool|float|int|string|Traversable|null $selection
+     * @param array $items
+     * @param array $tagOptions
+     *
+     * @psalm-param iterable<array-key, string>|string|\Stringable|int|float|bool|null $selection
+     * @psalm-param SelectItems $items
+     * @psalm-param SelectHtmlOptions $tagOptions
+     *
+     * @throws JsonException
+     *
+     * @return string
+     */
+    private static function renderSelectOptionTags($selection, array $items, array &$tagOptions): string
     {
         if (is_iterable($selection)) {
-            $selection = array_map('strval', (array)$selection);
+            $selection = array_map('strval', is_array($selection) ? $selection : iterator_to_array($selection));
```

Comment:
Originally it meant that if a single value is selected then you can specify it without wrapping it into array explicitly. Now it changed its meaning.